Public/Src/Tools/SymbolDaemon/Tool.SymbolDaemonRunner.dsc

```diff
@@ -144,13 +144,16 @@ function addFiles(createResult: SymbolCreateResult, args: OperationArguments, fi
 
     if (files.length === 0) {
         return undefined;
-    }               
+    }
+
+    const symbolMetadataFile = indexSymbolFiles(files);
 
     const fileMessageBody = files.length !== 0
         ? [
             Cmd.options("--file ", files.map(fi => Artifact.input(fi))),                

```

Comment:
>Artifact.input [](start = 51, length = 14)

Not totally sure how this works, just calling out for you to double check: For a normal Process pip this will specify the file as an input to ProcessBuilder which will cause the file to be materialized when the process is a cache miss. IPC pips are always a "miss". So just double checking that it doesn't have the side effect of having the files unconditionally materialized

Reply:
these files are inputs into: 1) indexingPip and 2) ipcPip

For (1) - these file must be materialized if we are going to index them, i.e., no cache hit
For (2) - these files are listed as inputs so pip can take dependency on them. In addition to that, these files are listed in lazilyMaterializedDependencies block. Adding elements to that block does not add dependency, it just populates a list of elements that should not be materialized.
For (2) cont'd - symbolMetadataFile is not on lazily materialized because Daemo is reading that file to get the metadata.

---
In reply to: [327711924](https://github.com/microsoft/BuildXL/pull/921#discussion_r327711924) [](ancestors = 327711924)